source/common/router/router.cc

```diff
@@ -1197,6 +1202,8 @@ void Filter::UpstreamRequest::onPoolReady(Http::StreamEncoder& request_encoder,
                                           Upstream::HostDescriptionConstSharedPtr host) {
   ENVOY_STREAM_LOG(debug, "pool ready", *parent_.callbacks_);
 
+  host->outlierDetector().putResult(Upstream::Outlier::Result::CONNECT_SUCCESS);
```

Comment:
This doesn't necessarily mean that a new connection was made since it may be using a preexisting connection. I'm not sure we want to account for this here? I guess you need this if you are going to tracking connection failures like you do above? Thoughts?

Reply:
I think I need it here, even when it may be using preexisting connection. Let us consider situation when there is 50 local origin errors (ex. port on upstream server was not opened). After that things are OK and subsequent 1000 requests are fine. Even when those 1000 requests use one preexisting connection, the ration of success to failures should be 1000 to 50, so we should count those successes here.